Assets/_BForBoss/_Character/Scripts/PlayerWallRunBehaviour.cs

```diff
@@ -81,50 +114,40 @@ public class PlayerWallRunBehaviour : MonoBehaviour
         #region PROPERTIES
         private Transform ChildTransform => _fpsCharacter != null ? _fpsCharacter.rootPivot : transform;
 
-        public bool IsWallRunning => _isWallRunning;
+        public bool IsWallRunning { get; private set; }
+
         #endregion
 
         #region PUBLIC_METHODS
-        public void Initialize(ECM2.Characters.Character baseCharacter, Func<Vector2> getMovementInput, Action<int> OnWallRunFinished)
+        public void Initialize(ECM2.Characters.Character baseCharacter, Func<Vector2> getMovementInput, Action<int> onWallRunFinished)
         {
             _baseCharacter = baseCharacter;
             _fpsCharacter = baseCharacter as FirstPersonPlayer;
             _movementInput = getMovementInput;
-            _OnWallRunFinished = OnWallRunFinished;
-            _mask = LayerMask.GetMask("ParkourWall");
+            _OnWallRunFinished = onWallRunFinished;
+            _parkourWallMask = LayerMask.GetMask(PARKOUR_WALL_LAYER);
         }
 
         public void Falling(Vector3 _)
```

Comment:
Wait, what is the point of this parameter? Also if it is needed, can we change its name to something more useful (even if it isn't being used)?

Reply:
The parameter doesn't have a purpose, which is why its just a discard character. This is kind of a pattern we've adopted for the custom movement behaviors (sliding, dashing, etc) where ECM2 has a method that we're overriding in `FirstPersonCharacter` and then the override just passes through to some custom logic in one of the custom behaviors. In this case, ECM2's falling state calls this every frame. The parameter is just there cause its a parameter of the character Falling method